bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewHubHandler.java

```diff
@@ -143,6 +147,21 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    private synchronized void pollShadesAsync() {
+        pollShadesOnceFuture = scheduler.submit(() -> {
+            try {
+                pollShades();
+            } catch (JsonParseException e) {
+                logger.warn("Bridge returned a bad JSON response: {}", e.getMessage());
+            } catch (HubProcessingException e) {
+                logger.warn("Error connecting to bridge: {}", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, e.getMessage());
```

Comment:
The detailed status is probably inappropriate.
If your bridge thing is really OFFLINE, if the thing handler is properly implemented, it will switch the thing status to OFLLINE / BRIDGE_OFFLINE elsewhere.
Maybe COMMUNICATION_ERROR is more appropriate ?

Reply:
It seems logical, yes. I've changed to COMMUNICATION_ERROR. But currently the only place where bridge if marked BRIDGE_OFFLINE is from method `poll` under the same conditions, i.e. as soon as any call would return HTTP code != 200, no response, network timeout etc. This seems similar, but I'm fine with letting only the polling job take responsibility for this.